examples/plugin/custom_sensor_system/OdometerSystem.cc

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <ignition/msgs/double.pb.h>
+
+#include <string>
+#include <unordered_map>
+#include <utility>
+
+#include <ignition/common/Profiler.hh>
+#include <ignition/plugin/Register.hh>
+#include <ignition/sensors/Noise.hh>
+#include <ignition/sensors/SensorFactory.hh>
+
+#include <sdf/Sensor.hh>
+
+#include <ignition/gazebo/components/CustomSensor.hh>
+#include <ignition/gazebo/components/Name.hh>
+#include <ignition/gazebo/components/ParentEntity.hh>
+#include <ignition/gazebo/components/Sensor.hh>
+#include <ignition/gazebo/components/World.hh>
+#include <ignition/gazebo/EntityComponentManager.hh>
+#include <ignition/gazebo/Util.hh>
+
+#include "Odometer.hh"
+#include "OdometerSystem.hh"
+
+using namespace custom;
+
+//////////////////////////////////////////////////
+void OdometerSystem::PreUpdate(const ignition::gazebo::UpdateInfo &,
+    ignition::gazebo::EntityComponentManager &_ecm)
+{
+  _ecm.EachNew<ignition::gazebo::components::CustomSensor,
+               ignition::gazebo::components::ParentEntity>(
+    [&](const ignition::gazebo::Entity &_entity,
+        const ignition::gazebo::components::CustomSensor *_custom,
+        const ignition::gazebo::components::ParentEntity *_parent)->bool
+      {
+        // Get sensor's scoped name without the world
+        auto sensorScopedName = ignition::gazebo::removeParentScope(
+            ignition::gazebo::scopedName(_entity, _ecm, "::", false), "::");
+        sdf::Sensor data = _custom->Data();
+        data.SetName(sensorScopedName);
+
+        // Default to scoped name as topic
+        if (data.Topic().empty())
+        {
+          std::string topic = scopedName(_entity, _ecm) + "/odometer";
```

Comment:
should we check here ig the topic is valid ?

Reply:
`ign-sensors` does that for us internally through the `SetTopic` function, so I think it's not necessary to do it here too:

https://github.com/ignitionrobotics/ign-sensors/blob/ba89178b6f4c75934ba608ab0ceb5348749b51f9/src/Sensor.cc#L218-L235

I do think that not all users would know that though